File: Codewars/Javascript/7kyu_SumOfABeach.js
/* Beaches are filled with sand, water, fish, and sun. Given a string, calculate how many times the words
"Sand", "Water", "Fish", and "Sun" appear without overlapping (regardless of the case).

Examples
sumOfABeach("WAtErSlIde")                    ==>  1
sumOfABeach("GolDeNSanDyWateRyBeaChSuNN")    ==>  3
sumOfABeach("gOfIshsunesunFiSh")             ==>  4 
sumOfABeach("cItYTowNcARShoW")               ==>  0 */

function sumOfABeach (beach) {
  let sandCount = (beach.match(/sand/ig) || []).length
  let waterCount = (beach.match(/water/ig) || []).length
  let fishCount = (beach.match(/fish/ig) || []).length
  let sunCount = (beach.match(/sun/ig) || []).length
  return sandCount + waterCount + fishCount + sunCount
}

File: Codewars/C#/7kyu_SumOfABeach.cs
/* Beaches are filled with sand, water, fish, and sun. Given a string, calculate how many times the words
"Sand", "Water", "Fish", and "Sun" appear without overlapping (regardless of the case).

Examples
SumOfABeach("WAtErSlIde")                    ==>  1
SumOfABeach("GolDeNSanDyWateRyBeaChSuNN")    ==>  3
SumOfABeach("gOfIshsunesunFiSh")             ==>  4
SumOfABeach("cItYTowNcARShoW")               ==>  0 */

using System;
using System.Text.RegularExpressions;

public class Kata
{
    public static int SumOfABeach(string s)
    {
        return Regex.Matches(s, "sand|water|fish|sun", RegexOptions.IgnoreCase).Count;
    }
}
